blockchain/utxobackend.go

```diff
@@ -352,15 +484,20 @@ func (l *LevelDbUtxoBackend) dbFetchUtxoStats(dbTx database.Tx) (*UtxoStats, err
 			// Ensure any deserialization errors are returned as database corruption
 			// errors.
 			if isDeserializeErr(err) {
-				str := fmt.Sprintf("corrupt utxo entry for %v: %v", outpoint, err)
-				return nil, makeDbErr(database.ErrCorruption, str)
+				return nil, AssertError(fmt.Sprintf("corrupt utxo entry for %v: %v",
+					outpoint, err))
 			}
 
 			return nil, err
 		}
 
 		stats.Total += entry.amount
 	}
+	iter.Release()
+	err := iter.Error()
```

Comment:
I think the iterator release should be deferred here considering there are a number of paths returning in the iteration.

Reply:
Good catch.  You are right that it should be released in the error paths as well.  Updated.